### PR TITLE
timg: 1.5.1 -> 1.5.2

### PR DESCRIPTION
**Pull request by hzeller**
Release notes:
https://github.com/hzeller/timg/releases/tag/v1.5.2

## Description of changes

Important: fixes a buffer overflow issue.

Also: slightly clean-up default.nix:
  * sort  dependencies
  * sort metadata attributes
  * use finalAttrs.
  * 
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/graphics/timg/default.nix [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
 , libsixel
 , openslide
+, pkg-config
+, stb
+, qoi
+, stdenv
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
```
Comment (drupol) [inline pkgs/tools/graphics/timg/default.nix:17]: Oh nice ! Thank you ! I don't feel alone to do that anymore!

Comment (hzeller) [reply pkgs/tools/graphics/timg/default.nix:17]: Yes, it was you pointing me to it in my [last PR](https://github.com/NixOS/nixpkgs/pull/245990) :)

File: pkgs/tools/graphics/timg/default.nix [hunk 2]
```diff
@@ -44,10 +48,10 @@ stdenv.mkDerivation rec {
   ];
 
   meta = with lib; {
```
[the remark this answers is not in the document]
Comment (hzeller) [reply pkgs/tools/graphics/timg/default.nix:50]: do you have an example somehwere of what you mean ?

[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/tools/graphics/timg/default.nix:50]: In here? https://github.com/NixOS/nixpkgs/pull/249725

[the remark this answers is not in the document]
Comment (hzeller) [reply pkgs/tools/graphics/timg/default.nix]: Alright, done! (sorry for the delay, currently at CCC-Camp)

[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/tools/graphics/timg/default.nix]: No hurry here :)

**Review by drupol — APPROVED** [review]
LGTM ! Let's wait the CI now...

**Pull request merged**